### PR TITLE
chore: Update to twilight 0.5

### LINKED ISSUE #22: Update `twilight-http` to 0.5
Update `twilight-http` to 0.5 in order to support slash commands (see issue #21).

### PR DESCRIPTION
**Pull request by Gelbpunkt**
Tested locally briefly (using /channels/id/messages), did not test the slash commands routes, assumed twilight-http is correct.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -215,23 +221,18 @@ async fn handle_request(
...
-        body,
-        form: None,
-        headers: Some(headers),
-        method,
-        path,
-        path_str: path_and_query,
-    };
+    let raw_request = RequestBuilder::raw(method, path, path_and_query)
+        .body(bytes)
+        .headers(headers.into_iter().map(|(k, v)| (k.unwrap(), v)))
```
Comment (Gelbpunkt) [inline src/main.rs]: I am not 100% sure what to do here. Should we return an error if there are two headers with the same name? (which is the only case when the key would be None)

Comment (spring4175) [reply src/main.rs]: We could probably just take the first. I can't think of a reason to send more than one, and IMO it's less of an error worth failing the request and more of a "why are you doing this? oh well."

**Comment by spring4175** [comment]
Marked as closing #22.

File: src/main.rs (hunk 1 above)
Comment (Gelbpunkt) [reply src/main.rs]: Alright, applied that

**Review by spring4175 — APPROVED** [review]

**Pull request merged**